### PR TITLE
Mitschriften Systementwicklung 25.05.2021

### PR DESCRIPTION
**Pull request by RvNovae**

### CODE REVIEW

**Comment by Steve2955** [comment]
## Command results
<details>
<summary>Details: </summary>

<details>
<summary><em>Run doctoc</em></summary>

```Shell
changed:
  - /home/runner/work/dhge-pi19-sem4/dhge-pi19-sem4/SWE-KASCHE/README.md
unchanged:
  - /home/runner/work/dhge-pi19-sem4/dhge-pi19-sem4/ABWL-LEDER/README.md
  - /home/runner/work/dhge-pi19-sem4/dhge-pi19-sem4/ASM-GUENTHER/README.md
  - /home/runner/work/dhge-pi19-sem4/dhge-pi19-sem4/ASM-GUENTHER/appendix/praktikum.md
  - /home/runner/work/dhge-pi19-sem4/dhge-pi19-sem4/DBS-DORENDORF/README.md
  - /home/runner/work/dhge-pi19-sem4/dhge-pi19-sem4/PRO-KASCHE/README.md
  - /home/runner/work/dhge-pi19-sem4/dhge-pi19-sem4/SWS-KUSCHE/README.md
  - /home/runner/work/dhge-pi19-sem4/dhge-pi19-sem4/EMB-GUENTHER/README.md
  - /home/runner/work/dhge-pi19-sem4/dhge-pi19-sem4/OOP-CASSELT/README.md
```



</details>

</details>

## Changed files
<details>
<summary>Changed file: </summary>

- SWE-KASCHE/README.md

</details>

<hr>

[:octocat: Repo](https://github.com/technote-space/toc-generator) | [:memo: Issues](https://github.com/technote-space/toc-generator/issues) | [:department_store: Marketplace](https://github.com/marketplace/actions/toc-generator)

File: SWE-KASCHE/README.md [hunk 1]
```diff
@@ -530,3 +533,157 @@ Mit Plantuml darstellen!
...
+
+class Konvertierer {
+  std::shared_ptr<Erbauer> e;
+
+  void setType(std::shared_ptr<Erbauer> f);
+};
+```
+
+**Singleton**
+- Ziel: nur genau ein Objekt einer Klasse soll existieren
```
Comment (RvNovae) [inline SWE-KASCHE/README.md:644]:
```suggestion

- Ziel: nur genau ein Objekt einer Klasse soll existieren
```

File: SWE-KASCHE/README.md [hunk 2]
```diff
@@ -530,3 +533,157 @@ Mit Plantuml darstellen!
...
+- Beispiel: Manager-Klasse zum Verwalten anderer Objekte
+- Spieler soll nur einmal existieren
+- Logger
+- Schnittstellenobjekt
+- Application-Context (Android)
+- Idee: private Constructors verwenden
+
```
Comment (RvNovae) [inline SWE-KASCHE/README.md:651]:
```suggestion

```

File: SWE-KASCHE/README.md [hunk 3]
```diff
@@ -530,3 +533,157 @@ Mit Plantuml darstellen!
...
+**Singleton**
+- Ziel: nur genau ein Objekt einer Klasse soll existieren
+- Beispiel: Manager-Klasse zum Verwalten anderer Objekte
+- Spieler soll nur einmal existieren
+- Logger
+- Schnittstellenobjekt
+- Application-Context (Android)
+- Idee: private Constructors verwenden
+  
+```cpp
```
Comment (RvNovae) [inline SWE-KASCHE/README.md:652]:
```suggestion

```cpp
```

File: SWE-KASCHE/README.md [hunk 4]
```diff
@@ -530,3 +533,157 @@ Mit Plantuml darstellen!
...
+  return GetInstance();
+}
+
+void main() {
+  singleton *s = Singleton.GetInstance();
+}
+```
+
+anstatt
+```cpp
```
Comment (RvNovae) [inline SWE-KASCHE/README.md:674]:
```suggestion

```cpp
```

File: SWE-KASCHE/README.md [hunk 5]
```diff
@@ -530,3 +533,157 @@ Mit Plantuml darstellen!
...
+private:
+  Singleton(Singleton &o const);
+```
```
Comment (RvNovae) [inline SWE-KASCHE/README.md:677]:
```suggestion
```

```

File: SWE-KASCHE/README.md [hunk 6]
```diff
@@ -530,3 +533,157 @@ Mit Plantuml darstellen!
...
+nun ($\copyright$ Scott Meyers)
+```cpp
```
Comment (RvNovae) [inline SWE-KASCHE/README.md:679]:
```suggestion

```cpp
```

**Comment by RvNovae** [comment]
"Multiple spaces after blockquote symbol"

da ist doch immer nur ein space 🤔

File: SWE-KASCHE/README.md [hunk 7]
```diff
@@ -530,3 +533,159 @@ Mit Plantuml darstellen!
...
+- View $\rightarrow$ Darstellung
+- Controller $\rightarrow$ Verwaltung
+- Model kommuniziert über den Controller mit der View (und anders herum)
+- Nutzen: oft hat man viele Views, die alle auf die gleichen Daten zugreifen $\rightarrow$ alle greifen über den Controller zu
+- Bsp: Tabellen
+
+**Entwurfsmuster**
+
+- $\rightarrow$ Erzeugende Entwurfsmuster
+  - > Prozess der Erzeugung
```
Comment (RvNovae) [inline SWE-KASCHE/README.md:555]:
```suggestion
  - Prozess der Erzeugung
```

**Review by nukerxy — APPROVED** [review]

**Review by ZeroPointMax — APPROVED** [review]
Vielen Dank

**Pull request merged**